### PR TITLE
feat : implement get all engagements with tests

### PR DESCRIPTION
**Pull request by sneh258**
Add get all engagements API
Add test cases for controllers and services
Add underscored as true in engagement model
Change folder naming convention

### CODE REVIEW

File: src/controllers/project.controller.js [hunk 1]
```diff
@@ -15,4 +15,18 @@ const getProject = async (req, res) => {
...
\ No newline at end of file
+const listProjects = async (req, res) => {
+  try {
+    const allProjects = await projectServices.listProjects();
+    res.status(200).json(allProjects);
+  } catch (error) {
+    {
+      res.status(500).json({
+        error: error.message
+
```
Comment (Chetan-MckinseyBEBR) [inline src/controllers/project.controller.js:26]: Remove unnecessary empty lines

File: src/services/project.services.js [hunk 2]
```diff
@@ -6,4 +6,10 @@ const getProject = async (id) => {
 
 };
 
-module.exports = { getProject };
\ No newline at end of file
+const listProjects = async () => {
+  const allProjects = await engagements.findAll();
+  return allProjects;
+
```
Comment (Chetan-MckinseyBEBR) [inline src/services/project.services.js:12]: Remove unnecessary empty lines in all ur file changes

File: tests/controllers/projects.controller.test.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
-const getProjectControllers = require('../../src/controllers/projectController');
-const getProjectServices = require('../../src/services/projectServices');
+const getProjectControllers = require('../../src/controllers/project.controller');
+const getProjectServices = require('../../src/services/project.services');
 
 describe('Engagements Controllers', () => {
```
Comment (Chetan-MckinseyBEBR) [inline tests/controllers/projects.controller.test.js:5]: Remove unnecessary empty lines in all ur file changes

File: tests/controllers/projects.controller.test.js [hunk 4]
```diff
@@ -58,6 +58,78 @@ describe('Engagements Controllers', () => {
...
+        engagement_id: 1223,
+        user_ids: ['1', '2', '3'],
+        case_study_ids: ['23', '34', '56']
+      }
+    ]
+    );
+    const mockReq = {};
+
+    const mockRes = {
+
```
Comment (Chetan-MckinseyBEBR) [inline tests/controllers/projects.controller.test.js:85]: Remove unnecessary empty lines

File: tests/services/projects.service.test.js [hunk 5]
```diff
@@ -25,4 +25,51 @@ describe('Engagement Services', () => {
       ]
     );
   });
+
+  it('should return list of all engagements from the database', async () => {
+    jest.spyOn(engagements, 'findAll').mockResolvedValue([
+
```
Comment (Chetan-MckinseyBEBR) [inline tests/services/projects.service.test.js:31]: Remove unnecessary empty lines

**Review by Chetan-MckinseyBEBR — COMMENTED** [review]
Added few comments

**Review by Chetan-MckinseyBEBR — APPROVED** [review]

**Review by Chetan-MckinseyBEBR — APPROVED** [review]

**Review by Chetan-MckinseyBEBR — APPROVED** [review]

File: src/services/project.services.js [hunk 6]
```diff
@@ -3,7 +3,11 @@ const { engagements } = require('../models');
 const getProject = async (id) => {
```
Comment (rohini-senthil) [inline src/services/project.services.js:3]: Can we rename this to `project.service.js`?

File: tests/controllers/projects.controller.test.js [hunk 7]
```diff
@@ -1,29 +1,22 @@
-const getProjectControllers = require('../../src/controllers/projectController');
```
Comment (rohini-senthil) [inline tests/controllers/projects.controller.test.js:1]: Can we rename the file to `project.controller.test.js`. Same with the other test file as well

File: src/routes/project.routes.js [hunk 8]
```diff
@@ -1,14 +1,10 @@
 const express=require('express');
```
Comment (rohini-senthil) [inline src/routes/project.routes.js:1]: Can we rename this `project.route.js`

**Review by rohini-senthil — APPROVED** [review]
Great work! 🚀

**Pull request merged**